### PR TITLE
propcheck: better check duplicated compilers/groups

### PR DESCRIPTION
**Pull request by dkm**
Check for duplicated compiler or group references:

group.cmipsel.compilers=cmips5el:cmipselg1210:cmips64elg1210
group.cmips64el.compilers=cmips564el:cmips64elg1210

triggers:
duplicated_compiler_references:
        cmips64elg1210

group.cmipss.compilers=&cmips:&cmipsel:&cmips64:&cmips64el
group.ccross.compilers=&cppcs:&cmipss:&cmips64el:&cnanomips:&cmrisc32:&cmsp:&cgccarm:&cavr:&rvcgcc:&cxtensaesp32:&cxtensaesp32s2:&cxtensaesp32s3:&cplatspec:&ckalray:&cs390x

triggers:
duplicated_group_references:
        cmips64el

Signed-off-by: Marc Poulhiès <dkm@kataplop.net>

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]
Nice! I hope that the script wasn't too hard to work with

Can you add two test cases to `etc/scripts/util/propschecktest.py` with the examples you give here? :)

**Comment by dkm** [comment]
Not issue working with the script, it's still simple enough for me to add stuff :)

File: etc/scripts/util/test/cases/bad_duplicated_compiler.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+compilers=&g1:&g2
+
+g1.compilers=c1:duplicatedname
+g2.compilers=c3:c4:duplicatedname
```
Comment (RubenRBS) [inline etc/scripts/util/test/cases/bad_duplicated_compiler.properties:4]: Newline at EOF

File: etc/scripts/util/test/cases/bad_duplicated_group.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+compilers=&g1:&g2
+
+g1.compilers=&dupgroup
+g2.compilers=&dupgroup
```
Comment (RubenRBS) [inline etc/scripts/util/test/cases/bad_duplicated_group.properties:4]: Newline at EOF

**Review by RubenRBS — APPROVED** [review]
Tyty, looks good

**Comment by mattgodbolt** [comment]
Oh my thanks so much for taking the time to improve this!

**Pull request merged**